src/transform/ir_proto_to_datalog/utils/access_path.h

```diff
@@ -0,0 +1,70 @@
+#ifndef SRC_TRANSFORM_IR_PROTO_TO_DATALOG_UTILS_ACCESS_PATH_H_
+#define SRC_TRANSFORM_IR_PROTO_TO_DATALOG_UTILS_ACCESS_PATH_H_
+
+#include "absl/strings/string_view.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+#include <vector>
+
+namespace raksha::transform::utils {
+
+// Represents an AccessPath descending through Manifest substructures. This
+// is eventually used for printing edges to the resultant Datalog file. It
+// does not own any strings to which it refers; it relies upon the original
+// Manifest tree structures generated from the original protos outliving it.
+//
+// Currently, this does not add type annotations onto the individual
+// components of the AccessPath. That may be wise to do in the future.
+class AccessPath {
+ public:
+  // Are two AccessPaths equal. Should be true exactly when they have
+  // equivalent string representations.
+  bool operator==(const AccessPath &other) const;
+
+  // Turns this AccessPath into a string representation. A path to the field
+  // f in the type T in particle P in recipe R will produce the string:
+  //
+  // "R.P.T.f"
+  std::string ToString() const;
+
+  // Create a leaf AccessPath from a string indicating a single string
+  // component. This will return an error status if
+  // check_component_allowed(leaf_component) does not return Status::Ok.
+  static absl::StatusOr<AccessPath> Create(absl::string_view leaf_component);
+
+  // Create a new AccessPath that is a parent of the parameter access_path by
+  // adding parent_component directly above the contents of access_path. If
+  // the string representation of access path is "a.b.c" and parent_component
+  // is "q", then the resulting access path would have the representation:
+  //
+  // "q.a.b.c"
+  //
+  // This will return an error status if
+  // check_component_allowed(parent_component) does not return Status::Ok.
+  static absl::StatusOr<AccessPath> CreateParent(
```

Comment:
Why `StatusOr`? If we have some structure to `AccessPath` as mentioned below, we could just use constructors directly?

Reply:
Good point. That would also allow us to validate the names as part of the deserialization process, which is really when you would want to catch issues like this.

Reply:
Done.